### PR TITLE
Fix HunyuanVideo produces NaN on PyTorch<2.5

### PR DESCRIPTION
**Pull request by hlky**
# What does this PR do?

`NaN` tracked to 

https://github.com/huggingface/diffusers/blob/811560b1d7daba48221317759ce0ed004513ea4f/src/diffusers/models/transformers/transformer_hunyuan_video.py#L117-L135

Specifically, some elements of `encoder_hidden_states`.

The dimensions of query, key, value and mask are large which suggests versions <2.5 used 32-bit indexing, this tracks with #10314 if ROCm versions are still using 32-bit indexing, this may also close that issue, awaiting confirmation from user.

Tested on CUDA 2.4.1

https://github.com/user-attachments/assets/63b05836-e9a9-4f4d-9169-d71194b5e071

<details>
  <summary>Code</summary>

  ```python
import torch
from diffusers import HunyuanVideoPipeline, HunyuanVideoTransformer3DModel
from diffusers.utils import export_to_video

model_id = "hunyuanvideo-community/HunyuanVideo"
transformer = HunyuanVideoTransformer3DModel.from_pretrained(
    model_id, subfolder="transformer", torch_dtype=torch.bfloat16
)
pipe = HunyuanVideoPipeline.from_pretrained(model_id, transformer=transformer, torch_dtype=torch.float16).to("cuda")
pipe.vae.enable_tiling()

output = pipe(
    prompt="A cat walks on the grass, realistic",
    height=320,
    width=512,
    num_frames=61,
    num_inference_steps=30,
).frames[0]
export_to_video(output, "output.mp4", fps=15)
```
</details>

There's also a small performance increase

| 2.4.1 with fix | 2.5.1 | 2.5.1 with fix |
|--------|--------|--------|
| 30/30 [01:56<00:00,  3.88s/it] | 30/30 [02:04<00:00,  4.16s/it] | [01:56<00:00,  3.89s/it] | 

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sayakpaul @a-r-r-o-w

### CODE REVIEW

**Comment by Nerogar** [comment]
This change broke batching again. It was previously fixed in #10454
